crypto/fipsmodule/aes/asm/aesni-avx512.pl

```diff
@@ -0,0 +1,3165 @@
+      $code .= "vaesdec $t0, $st[$i], $st[$i]\n";
+    }
+
+    # round 11
+    $code .= "vbroadcasti32x4 0x130($TW), $t0\n";
+    for (my $i = 0; $i < 4; $i++) {
+      $code .= "vaesdec $t0, $st[$i], $st[$i]\n";
+    }
+
+    # round 12
+    $code .= "vbroadcasti32x4 0x140($TW), $t0\n";
+    for (my $i = 0; $i < 4; $i++) {
+      $code .= "vaesdec $t0, $st[$i], $st[$i]\n";
+    }
+
+    # round 13
+    $code .= "vbroadcasti32x4 0x150($TW), $t0\n";
+    for (my $i = 0; $i < 4; $i++) {
+      $code .= "vaesdec $t0, $st[$i], $st[$i]\n";
+    }
+
+    # round 14
+    $code .= "vbroadcasti32x4 0x160($TW), $t0\n";
+    for (my $i = 0; $i < 4; $i++) {
+      $code .= "vaesdeclast $t0, $st[$i], $st[$i]\n";
+    }
+
+    # xor Tweak values
+    for (my $i = 0; $i < 4; $i++) {
+      $code .= "vpxorq    $tw[$i], $st[$i], $st[$i]\n";
+    }
+
+    $code .= <<___;
+    # load next Tweak values
+    vmovdqa32  %zmm15, $tw[0]
+    vmovdqa32  %zmm16, $tw[1]
+    vmovdqa32  %zmm17, $tw[2]
+    vmovdqa32  %zmm18, $tw[3]
+___
+  }
+
+  # ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+  # ;void aes_hw_xts_encrypt_avx512(
+  # ;               const uint8_t *in,        // input data
+  # ;               uint8_t *out,             // output data
+  # ;               size_t length,            // sector size, in bytes
+  # ;               const AES_KEY *key1,      // key used for "ECB" encryption, 16*2 bytes
+  # ;               const AES_KEY *key2,      // key used for tweaking, 16*2 bytes
+  # ;               const uint8_t iv[16])      // initial tweak value, 16 bytes
+  # ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+  my $rndsuffix = &random_string();
+
+  $code .= ".text\n";
+
+  {
+  $code.=<<___;
+  .globl	aes_hw_xts_encrypt_avx512
+  .hidden	aes_hw_xts_encrypt_avx512
+  .type	aes_hw_xts_encrypt_avx512,\@abi-omnipotent
+  .align	32
+  aes_hw_xts_encrypt_avx512:
+  .cfi_startproc
+          endbranch
+___
+  }
+  $code .= "push 	 %rbp\n";
+  $code .= "mov 	 %rsp,%rbp\n";
+  $code .= "sub 	 \$$VARIABLE_OFFSET,%rsp\n";
+  $code .= "and 	 \$0xffffffffffffffc0,%rsp\n";
+  $code .= "mov 	 %rbx,$GP_STORAGE($TW)\n";
+
+  if ($win64) {
+    $code .= "mov 	 %rdi,$GP_STORAGE + 8*1($TW)\n";
+    $code .= "mov 	 %rsi,$GP_STORAGE + 8*2($TW)\n";
+    $code .= "vmovdqa      %xmm6, $XMM_STORAGE + 16*0($TW)\n";
+    $code .= "vmovdqa      %xmm7, $XMM_STORAGE + 16*1($TW)\n";
+    $code .= "vmovdqa      %xmm8, $XMM_STORAGE + 16*2($TW)\n";
+    $code .= "vmovdqa      %xmm9, $XMM_STORAGE + 16*3($TW)\n";
+    $code .= "vmovdqa      %xmm10, $XMM_STORAGE + 16*4($TW)\n";
+    $code .= "vmovdqa      %xmm11, $XMM_STORAGE + 16*5($TW)\n";
+    $code .= "vmovdqa      %xmm12, $XMM_STORAGE + 16*6($TW)\n";
+    $code .= "vmovdqa      %xmm13, $XMM_STORAGE + 16*7($TW)\n";
+    $code .= "vmovdqa      %xmm14, $XMM_STORAGE + 16*8($TW)\n";
+    $code .= "vmovdqa      %xmm15, $XMM_STORAGE + 16*9($TW)\n";
+  }
+
+  $code .= "mov 	 \$0x87, $ghash_poly_8b\n";
```

Comment:
I think since `ghash` is not the algorithm used here, we can replace the occurrences of `ghash` with `gf` or, preferably, `GF`.

Reply:
will replace ghash_poly_8b to gf_poly_8b, same for others